# **Business Understanding**

## **Project Overview**
The company is expanding into the aviation industry and plans to purchase and operate airplanes for commercial and private enterprises. However, it currently lacks knowledge about the relative safety and risks associated with different aircraft.

## **Business Problem**
The key challenge is determining which aircraft are **lowest risk** for purchase and operation. Making uninformed decisions could lead to increased accident rates, financial loss, and reputational damage.

**Key Business Question:**  
Which aircraft models and manufacturers present the *lowest operational risk* based on historical accident and safety data?

## **Stakeholders**
- **Head of the Aviation Division:** Needs actionable insights to decide which aircraft to invest in.  
- **Company Executives:** Interested in diversifying the company portfolio while minimizing operational risk.  
- **Operations Team:** Requires guidance for safe aircraft selection and planning.

## **Project Goals**
- Identify patterns in aviation accident data to determine aircraft safety.  
- Analyze historical accident trends to evaluate risks by aircraft type, manufacturer, and usage.  
- Provide three evidence-based recommendations to guide aircraft purchase decisions.

## **Analytical Approach**
This project will analyze historical aviation accident data to identify low-risk aircraft. The analysis involves exploring accident trends, calculating risk metrics (e.g., fatality rates), and visualizing patterns. Insights will support evidence-based decision-making in aircraft selection.

## **Value to the Stakeholder**
This project will enable the aviation division to:  
- Make **data-driven decisions** about aircraft acquisition.  
- Minimize **financial and operational risks** associated with unsafe aircraft.  
- Prioritize aircraft investments that maximize safety and reliability.

## **Success Criteria**
The project is successful if it identifies at least three aircraft models with significantly lower fatality or accident rates compared to the fleet average.

# **Data Understanding**

## **Overview**
The dataset comes from the **National Transportation Safety Board (NTSB)** and contains information about **civil aviation accidents** and **selected incidents** from **1962 to 2023**.  
Each record represents an individual aviation event and includes details such as:
- **Date and location of the accident**
- **Aircraft type, make, and model**
- **Flight purpose**
- **Number of injuries and fatalities**
- **Weather conditions**
- **Investigation findings**

The goal of this step is to explore the dataset, understand its structure, and assess data quality before analysis.

## ** Importing Libraries and Loading Data**

In [19]:
# Import necessary libraries
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings('ignore')

# Load dataset
df = pd.read_csv("../data/Aviation_Data.csv")


# Display basic information
print("Dataset shape:", df.shape)
df.head()

Dataset shape: (90348, 31)


Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country Latitude Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States      NaN       NaN          NaN   
1   BRIDGEPORT, CA  United States      NaN       NaN          NaN   
2    Saltville, VA  United States  36.9222  -81.8781          NaN   
3       EUREKA, CA  United States      NaN       NaN          NaN   
4       Canton, OH  United States      NaN       NaN          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   
3          NaN  ...          Personal         NaN                  2.0   
4          NaN  ...          Personal         NaN                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  
3               IMC                 Cruise  Probable Cause       12-09-2000  
4               VMC               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

## **2. Dataset Structure**
We’ll review column names, data types, and non-null counts to understand the composition of the dataset.

In [20]:
# Inspect dataset structure
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90348 entries, 0 to 90347
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      90348 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50249 non-null  object 
 9   Airport.Name            52790 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87572 non-null  object 
 14  Make                    88826 non-null

## **3. Preview of Data**
Let’s view the first few rows to understand what the dataset looks like.

In [21]:
# View first 5 records
df.head()

Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country Latitude Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States      NaN       NaN          NaN   
1   BRIDGEPORT, CA  United States      NaN       NaN          NaN   
2    Saltville, VA  United States  36.9222  -81.8781          NaN   
3       EUREKA, CA  United States      NaN       NaN          NaN   
4       Canton, OH  United States      NaN       NaN          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   
3          NaN  ...          Personal         NaN                  2.0   
4          NaN  ...          Personal         NaN                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  
3               IMC                 Cruise  Probable Cause       12-09-2000  
4               VMC               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

## **4. Missing Values Assessment**
Missing data can distort results, so we’ll examine which columns have missing values and how much data is affected.

In [22]:
# Count missing values per column
missing_data = df.isnull().sum().sort_values(ascending=False)
missing_percentage = (missing_data / len(df)) * 100

missing_df = pd.DataFrame({
    'Missing Values': missing_data,
    'Percentage (%)': missing_percentage.round(2)
})
missing_df.head(15)

Missing Values  Percentage (%)
Schedule                         77766           86.07
Air.carrier                      73700           81.57
FAR.Description                  58325           64.56
Aircraft.Category                58061           64.26
Longitude                        55975           61.95
Latitude                         55966           61.94
Airport.Code                     40099           44.38
Airport.Name                     37558           41.57
Broad.phase.of.flight            28624           31.68
Publication.Date                 16689           18.47
Total.Serious.Injuries           13969           15.46
Total.Minor.Injuries             13392           14.82
Total.Fatal.Injuries             12860           14.23
Engine.Type                       8536            9.45
Report.Status                     7840            8.68

## **5. Unique Values and Categorical Insights**
Let’s explore categorical columns (e.g., Aircraft Type, Make, Model, Purpose of Flight) to see how diverse the dataset is.

In [23]:
# Check unique value counts for key categorical columns
categorical_cols = ['Make', 'Model', 'Broad_Phase_of_Flight', 'Purpose_of_Flight', 'Weather_Condition']
for col in categorical_cols:
    if col in df.columns:
        print(f"{col}: {df[col].nunique()} unique values")
        print(df[col].value_counts().head(5), "\n")

Make: 8237 unique values
Cessna    22227
Piper     12029
CESSNA     4922
Beech      4330
PIPER      2841
Name: Make, dtype: int64 

Model: 12318 unique values
152          2367
172          1756
172N         1164
PA-28-140     932
150           829
Name: Model, dtype: int64 



## **6 Summary Statistics**
We’ll check for numeric summaries to understand the range and central tendency of numerical features (e.g., total injuries, fatalities).

In [24]:
# Summary statistics for numeric columns
df.describe()

Number.of.Engines  Total.Fatal.Injuries  Total.Serious.Injuries  \
count       82805.000000          77488.000000            76379.000000   
mean            1.146585              0.647855                0.279881   
std             0.446510              5.485960                1.544084   
min             0.000000              0.000000                0.000000   
25%             1.000000              0.000000                0.000000   
50%             1.000000              0.000000                0.000000   
75%             1.000000              0.000000                0.000000   
max             8.000000            349.000000              161.000000   

       Total.Minor.Injuries  Total.Uninjured  
count          76956.000000     82977.000000  
mean               0.357061         5.325440  
std                2.235625        27.913634  
min                0.000000         0.000000  
25%                0.000000         0.000000  
50%                0.000000         1.000000  
75%                0.000000         2.000000  
max              380.000000       699.000000

## **7. Initial Observations**

From the exploration, we can summarize the following key points:

- The dataset contains **31 columns and 90348 rows** ,representing aviation accidents and incidents spanning **over 60 years**.  
- There are missing values** across the dataset, with some columns (e.g., `Weather_Condition`, `Broad_Phase_of_Flight`) containing a significant portion of them.  
- The dataset includes a **large variety of aircraft makes and models**, which will be valuable when comparing accident rates and assessing risk levels.  
- Date and numeric fields (e.g., injury counts) may require **format standardization and conversion** before analysis.  
- Some columns appear irrelevant or inconsistent and may be dropped in the **Data Preparation** phase.  

Next, we’ll clean and prepare this dataset to ensure it’s consistent and ready for detailed analysis.

# **Data Preparation**

## **Overview**
After understanding the dataset’s structure and quality, the next step is **Data Preparation**.  
This involves cleaning, formatting useful features to ensure the dataset is consistent and reliable for analysis.

The main goals of this step are to:
1. Handle missing values  
2. Fix incorrect data types  
3. Select relevant columns  
4. Create new features for risk analysis  
5. Ensure the dataset is ready for visualization and modeling

## **1. Removing Duplicates**
Duplicate entries can distort statistics, so we’ll identify and remove them.

In [26]:
# Count and remove duplicates
duplicates = df.duplicated().sum()
print(f"Duplicate rows found: {duplicates}")

df.drop_duplicates(inplace=True)
print("New dataset shape:", df.shape)

Duplicate rows found: 1390
New dataset shape: (88958, 31)
